store/store.go

```diff
@@ -159,30 +165,68 @@ func (s *Store) Put(key []byte, value []byte) error {
 		return err
 	}
 
-	has, err := s.Has(key)
+	// Get the key in primary storage
+	indexKey, err := s.index.Primary.IndexKey(key)
 	if err != nil {
 		return err
 	}
-	// we assume that a second put should NOT write twice
```

Comment:
rather than delete this code, can we make an option when you setup a store that allows you to specify whether a store supports updates? 

For a blockstore, a blockstore has no update, and I'm concerned this code changes behavior expected by HashedBlockstore.Put. When the hash is a function of the data itself, the code we're deleting here IS the correct behavior -- blocks are immutable, and you should never be writing a block with the same hash twice.

Reply:
*priority*

Reply:
We cover this case in Put. Instead of doing a `Has` here we `Get` and check the key and the value, if they are both equal we do nothing (which is the case of the Blockstore), so no behavior is changed. See: https://github.com/adlrocha/go-storethehash/blob/6798e4fc45299553187ffb75f141f2f1b1c33d9a/store/store.go#L189 (I updated this code and now we return the right error, I'll push it in a moment).

However, if you feel it is worth having an option to enable storethehash as an immutable store (as a sanity-check), we can also do that.

Reply:
I think we should have the option for storethehash as an immutable store -- since blockstore is a big future use case. hopefully that doesn't become a burden to maintain. maybe it becomes a fork.